### PR TITLE
UIDEXP-130: Implement search feature in mapping profile transformations

### PR DESCRIPTION
**Pull request by VictorSoroka1**
## Purpose

Implement search feature in mapping profile transformations in the scope of the [UIDEXP-130](https://issues.folio.org/browse/UIDEXP-130).

## Screenshot
Screen Shot 2020-08-03 at 5 12 32 PM

### CODE REVIEW

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/MappingProfilesTransformationsModal.js [hunk 1]
```diff
@@ -36,18 +40,29 @@ export const MappingProfilesTransformationsModal = ({
...
     if (!isOpen) {
       setFilterPaneOpen(true);
-      setSearchFilters(initialSearchFormValues.filters);
+      resetSearchForm();
     }
-  }, [isOpen]);
+  }, [isOpen, resetSearchForm]);
 
-  const searchResults = initialTransformationsValues.transformations
+  const searchValueResults = !searchValue
```
Comment (YevheniiMaltsev) [inline src/settings/MappingProfiles/MappingProfilesTransformationsModal/MappingProfilesTransformationsModal.js:60]: It looks cool.

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js [hunk 2]
```diff
@@ -73,25 +72,25 @@ export const TransformationField = ({
...
-          visibleColumns={visibleColumns}
-          formatter={formatter}
-          autosize={autosize}
-        />
+        <div className={css.mappingProfilesFormTransformations}>
+          <MultiColumnList
+            id="mapping-profiles-form-transformations"
+            fields={fields}
+            contentData={contentData || fields.value}
+            totalCount={fields.value.length}
```
Comment (YevheniiMaltsev) [inline src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js:80]: I wonder, if we use search result (For example, we display only 2 of 10 items), so should we apply the number of all items here in this case? I'm not sure but property can used to calculate the list height so this is why we have to apply some additional styles to fix it. @VictorSoroka1 what do you think about it?

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js [hunk 3]
```diff
@@ -73,25 +72,25 @@ export const TransformationField = ({
...
-          columnWidths={columnWidths}
-          visibleColumns={visibleColumns}
-          formatter={formatter}
-          autosize={autosize}
-        />
+        <div className={css.mappingProfilesFormTransformations}>
+          <MultiColumnList
+            id="mapping-profiles-form-transformations"
+            fields={fields}
+            contentData={contentData || fields.value}
```
Comment (YevheniiMaltsev) [inline src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js:79]: As far as I understand fields.value will be applied will be used if search and filter result is empty. Please, could you clarify why do we need to apply fields.value in this case?

Comment (VictorSoroka1) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js:79]: Form relates on fields.value in every case. If nothing is provided the form will be empty. fields.value is a source of truth for the form. `contentData` is used for now as the approach for the fields form outside the modal.

Comment (YevheniiMaltsev) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js:79]: I guess, contentData is used inside of the modal. But I got your idea, thanks for the clarificaiton

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js (hunk 2 above)
Comment (VictorSoroka1) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js:80]: Regarding showing of a number of total items - I looked through the experience of how it is done in other places and found that only the amount of found items is shown. We can suggest this as a further improvement to the PO on grooming.

Regarding the height, I would not use the number of items for calculation of height as the height of each row can vary if there is too much content in the row.

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js (hunk 3 above)
Comment (VictorSoroka1) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js:79]: You are right. Vice verse to what I wrote, field.value is needed for now for the case outside modal. Should be removed after we get rid of it. Will add comment per your suggestion. Thanks.

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/MappingProfilesTransformationsModal.js (hunk 1 above)
Comment (VictorSoroka1) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/MappingProfilesTransformationsModal.js:60]: Agree, I knew that you would pay attention to this :)

File: src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js (hunk 2 above)
Comment (VictorSoroka1) [reply src/settings/MappingProfiles/MappingProfilesTransformationsModal/TransformationsField/TransformationsField.js]: Turned out styles can be removed as the `autozise` prop for MCL can now be removed after the search pane visibility is manged via CSS. `autozise` was used solely to handle the case when the search pane hides and thus the MCL recalculates the column widths. The way `autozise` does it is by setting the width and height on the mcl container which should be overridden in our case for the wrong calculation of height when the list renders not all possible transformations rows. For some reason, it worked incorrectly when search pane was hidden via conditional rendering in the component. Thanks for the comment which allowed me to rethink this idea once again.

**Pull request merged**